### PR TITLE
Log information about DW DB connection pool connections and blocking

### PR DESCRIPTION
**Pull request by jeff303**
Add new namespace to log diagnostic info about the DW DB connection pool from execute, when a connection is requested

Updating default log4j2.xml to capture these messages (which are at DEBUG level)

### CODE REVIEW

**Comment by jeff303** [comment]
Example snippet that shows how this log message is interleaved with the existing one.

```
2021-04-19 21:42:43,852 DEBUG middleware.log :: POST /api/card/7/query 202 [ASYNC: completed] 827.4 ms (26 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (40 idle, 0 queued) (225 total active threads) Queries in flight: 22 (0 queued)
2021-04-19 21:42:43,871 DEBUG middleware.log :: POST /api/card/7/query 202 [ASYNC: completed] 534.6 ms (26 DB calls) App DB connections: 2/15 Jetty threads: 4/50 (40 idle, 0 queued) (225 total active threads) Queries in flight: 22 (0 queued)
2021-04-19 21:42:43,901 DEBUG middleware.log :: POST /api/card/7/query 202 [ASYNC: completed] 898.2 ms (26 DB calls) App DB connections: 3/15 Jetty threads: 4/50 (40 idle, 0 queued) (225 total active threads) Queries in flight: 22 (0 queued)
2021-04-19 21:42:43,954 DEBUG execute.pool-diagnostic :: DB 2 connection needed: 3 active of 3 total, 17 threads waiting
2021-04-19 21:42:43,969 DEBUG execute.pool-diagnostic :: DB 2 connection needed: 3 active of 3 total, 18 threads waiting
2021-04-19 21:42:43,987 DEBUG execute.pool-diagnostic :: DB 2 connection needed: 3 active of 3 total, 19 threads waiting
```

So this effectively doubles the number of log messages.  If we want to get that into the same line, I would need some guidance on how to thread the necessary info out of the query layer and into the logging middleware.

**Comment by jeff303** [comment]
OK, I reworked things so that the info could all be captured in the existing message, rather than a duplicate one.  Here are some examples (under heavy load, then at the end of load):

```
2021-04-20 17:40:58,810 DEBUG middleware.log :: POST /api/card/7/query 200 [ASYNC: completed] 1.3 s (26 DB calls) App DB connections: 2/15 Jetty threads: 4/50 (41 idle, 0 queued) (228 total active threads) Queries in flight: 50 (8 queued); postgres DB 2 connections: 3/3 (45 threads blocked)
2021-04-20 17:41:19,043 DEBUG middleware.log :: POST /api/card/7/query 200 [ASYNC: completed] 258.1 ms (26 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (41 idle, 0 queued) (228 total active threads) Queries in flight: 0 (0 queued); postgres DB 2 connections: 2/3 (0 threads blocked)
```

File: src/metabase/driver/sql_jdbc/execute.clj [hunk 1]
```diff
@@ -133,6 +135,15 @@
   ^DataSource [database]
   (:datasource (sql-jdbc.conn/db->pooled-connection-spec database)))
```
Comment (jeff303) [inline src/metabase/driver/sql_jdbc/execute.clj:136]: Is it safe to assume this will always return an instance of `PoolBackedDataSource` ?

Comment (jeff303) [reply src/metabase/driver/sql_jdbc/execute.clj:136]: @camsaul , could use your eyes on this one; thanks.

**Review by robdaemon — APPROVED** [review]
from a functionality perspective, this gets a 👍🏻 from me. I'll leave code review to Cam or others on the core team

**Pull request merged**